cpp/src/strings/json/json_path.cu

```diff
@@ -217,19 +177,23 @@ class json_state : private parser {
       parent_el_type(json_element_type::NONE)
   {
   }
-  __device__ json_state(const char* _input, int64_t _input_len)
+  __device__ json_state(const char* _input,
+                        int64_t _input_len,
+                        thrust::optional<get_json_object_options> _options)
     : parser(_input, _input_len),
       cur_el_start(nullptr),
       cur_el_type(json_element_type::NONE),
       parent_el_type(json_element_type::NONE)
   {
+    options = _options.value_or(get_json_object_options{});
```

Comment:
You pass an optional object which a default value, then you also get the object from the optional with a default fallback. Can we totally get rid of optional and use a default object at the beginning?